build/quicklink.php

```diff
@@ -132,3 +170,37 @@ function quicklink_plugin_compatibility_files() {
 	}
 }
 add_action( 'init', 'quicklink_plugin_compatibility_files' );
+
+/**
+ * Add quicklink to the default scripts to make it available earlier in the runtime.
+ *
+ * @param WP_Scripts $scripts The WP_Scripts instance.
+ *
+ * @return void
+ */
+function quicklink_to_default_scripts( $scripts ) {
+	$scripts->add( 'quicklink', QUICKLINK_URL . 'quicklink.min.js', array(), '<##= pkg.version ##>', true );
+}
+add_action( 'wp_default_scripts', 'quicklink_to_default_scripts' );
+
+/**
+ * Prefetch a url with quicklink.
+ *
+ * @param  string  $url         The URL to be prefetched.
+ * @param  boolean $is_priority Whether or not the URL should be treaded as high priority target.
+ *
+ * @return void
+ */
+function quicklink_prefetch( $url, $is_priority = false ) {
+	add_filter(
+		'quicklink_prefetch',
+		function( $prefetch ) use ( $url, $is_priority ) {
+			$prefetch[] = array(
+				'url'        => $url,
+				'isPriority' => $is_priority,
+			);
+
+			return $prefetch;
+		}
+	);
```

Comment:
```suggestion
	wp_add_inline_script(
		'quicklink',
		sprintf( 'quicklink.prefetch( %s, %s );', wp_json_encode( $url ), wp_json_encode( $is_priority ) ),
		'after'
	);
```

Reply:
We might generate a timing issue here. 

Quicklink is loaded in the footer for performance reasons. Calling quicklink in the body would currently produce an error as the interpreter is calling the function before it is available. Thats why I'm currently building the queue.

Reply:
The `wp_add_inline_script()` function will ensure the provided JS code is printed after the Quicklink script is printed on the footer. So I don't see a timing issue.

Reply:
Nevertheless, this could also be guarded against by wrapping it in a `load` event handler:

```suggestion
	wp_add_inline_script(
		'quicklink',
		sprintf( 'addEventListener( 'load', function() { quicklink.prefetch( %s, %s ); }', wp_json_encode( $url ), wp_json_encode( $is_priority ) ),
		'after'
	);
```

This would be better than my suggestion above anyway since it would prevent prefetching from slowing down the page load, potentially.

Also, it would have another benefit: we could add a `script_loader_tag` filter which injects the `async` attribute onto the `script` which loads `quicklink`. This would improve performance yet further by preventing the script from causing the page to block rendering.

Reply:
Ah, the load parameter actually makes sense. 

Personally I would much rather have the filter option for the sole fact to make the queue filterable (which is still possible, but a bit more complicated now), but this seems to be the more elegant solution.